pkg/restapi/issuer/operation/operations.go

```diff
@@ -1222,25 +1222,46 @@ func (o *Operation) saveCredentialAttachmentData(thID string, credOffered creden
 
 func (o *Operation) readAndValidateCredentialApplication(msg service.DIDCommAction,
 	credManifest *cm.CredentialManifest) error {
-	// reading credential application if issuer have sent, out presentation_definition along with manifest
-	// TODO - Aries Validate signatures and proofs of credential application should be part of
-	// cm.ValidateCredentialApplicationAttachment function.
 	if credManifest.PresentationDefinition != nil {
 		applicationAttachments, err := getAttachments(msg)
 		if err != nil {
 			return fmt.Errorf("failed to get request credential attachments: %w", err)
 		}
 
-		if len(applicationAttachments) != 1 {
-			return errors.New("invalid request credential message, expected valid credential application")
+		attachmentAsMap, ok := applicationAttachments[0].Data.JSON.(map[string]interface{})
```

Comment:
Can't we use `applicationAttachments[0].Data.JSON` and unmarshaling than type casting to map ?

Reply:
updated